File: AstryxSnapshot.java
package astryx.spine;

import java.util.Map;

public class AstryxSnapshot {
    public String agent_name;
    public String identity_reason;
    public String user_input;
    public Map<String, Object> perception;
    public String goal;
    public Map<String, Object> plan;
    public String reply;
    public StateSnapshot state_snapshot;

    public static class StateSnapshot {
        public double mood;
        public Map<String, Double> traits;
    }
}

File: AstryxSnapshot.cs
using System.Collections.Generic;

namespace ASTRYX.Dashboard.Models
{
    public class AstryxSnapshot
    {
        public string agent_name { get; set; }
        public string identity_reason { get; set; }
        public string user_input { get; set; }
        public Dictionary<string, object> perception { get; set; }
        public string goal { get; set; }
        public Dictionary<string, object> plan { get; set; }
        public string reply { get; set; }
        public StateSnapshot state_snapshot { get; set; }
    }

    public class StateSnapshot
    {
        public double mood { get; set; }
        public Dictionary<string, double> traits { get; set; }
    }
}